modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -100,13 +100,14 @@
   protected ExtensionComponent(ExtensionModel extensionModel,
                                ComponentModel componentModel,
                                ConfigurationProvider configurationProvider,
-                               ExtensionManagerAdapter extensionManager) {
+                               ExtensionManagerAdapter extensionManager,
+                               MetadataMediator metadataMediator) {
```

Comment:
why is it needed to expose this in the constructor? Why can't each component know how to build its own mediator?